internal/components/receivers_test.go

```diff
@@ -99,6 +99,15 @@ func TestDefaultReceivers(t *testing.T) {
 			receiver:     "awsxray",
 			skipLifecyle: true, // Requires AWS endpoint to check identity to run
 		},
+		{
+			receiver: "azureblob",
+			getConfigFn: func() component.Config {
+				cfg := rcvrFactories["azureeventhub"].CreateDefaultConfig().(*azureeventhubreceiver.Config)
+				cfg.Connection = "DefaultEndpointsProtocol=https;AccountName=accountName;AccountKey=accountKey==;EndpointSuffix=core.windows.net"
+				return cfg
+			},
+			skipLifecyle: true, // Requires Azure blob store to run
```

Comment:
Is there a way to configure this component that it can use a mock server or mock body so it can run within the lifecycle checks?

Reply:
The best example I have is `exporter/awskinesisexporter` which would give some idea on how I've used mocks in the past.

Reply:
no other tests seem to do this so I'm hesitant to bend the test suite to such an extent just for one test

Reply:
I took a closer look at the code. The actual issue is that it tries to connect to an Azure event hub over amqps. The URL is hardcoded to use TLS here: https://github.com/Azure/azure-amqp-common-go/blob/master/conn/conn.go#L54 and therefore can't be mocked (there doesn't seem to be any way to turn of certificate checking.

Therefore, I don't think it's possible to mock this interaction and I think I am forced to leave skipLifecycle to true.

Reply:
No worries, sorry I forgot to follow up here.